src/core/public/keyboard_shortcuts/keyboard_shortcuts_service.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { KeyboardShortcutsSetup, KeyboardShortcutsStart, ShortcutDefinition } from './types';
+
+/** @internal */
+export class KeyboardShortcutsService {
+  // Phase 1.6: Basic Registration System - Simple shortcut storage
+  private shortcuts = new Map<string, ShortcutDefinition>();
+  private isListening = false;
+
+  // Phase 1.2: Basic Service Class Structure
+  public setup(): KeyboardShortcutsSetup {
+    return {
+      register: (shortcuts) => this.register(shortcuts),
+    };
+  }
+
+  public start(): KeyboardShortcutsStart {
+    // Phase 1.4: Basic Event Listener Setup
+    this.startEventListening();
+
+    return {
+      register: (shortcuts) => this.register(shortcuts),
+      unregister: (id) => this.unregister(id),
+    };
+  }
+
+  public stop() {
+    this.stopEventListening();
+    this.shortcuts.clear();
+  }
+
+  // Phase 1.6: Basic Registration System
+  private register(shortcuts: ShortcutDefinition[]): string[] {
+    const registeredIds: string[] = [];
+
+    shortcuts.forEach((shortcut) => {
+      // Construct full ID: shortcut.id + "." + shortcut.pluginId
+      const fullId = `${shortcut.id}.${shortcut.pluginId}`;
+
+      // Basic storage - no validation in Phase 1
+      this.shortcuts.set(shortcut.keys.toLowerCase(), shortcut);
+      registeredIds.push(fullId);
+    });
+
+    return registeredIds;
+  }
+
+  private unregister(fullId: string): void {
+    // Find and remove shortcut by full ID
+    for (const [key, shortcut] of this.shortcuts.entries()) {
+      if (`${shortcut.id}.${shortcut.pluginId}` === fullId) {
+        this.shortcuts.delete(key);
+        return;
+      }
+    }
+  }
+
+  // Phase 1.5: Input Context Filtering - Text input detection
+  private isTextInputActive(target: EventTarget | null): boolean {
+    if (!target || !(target as HTMLElement).tagName) return false;
+
+    const element = target as HTMLElement;
+    const tagName = element.tagName.toLowerCase();
```

Comment:
lets not toLowerCase it. Since browsers capitalize all by convention, lets follow that convention in the code